### DISCUSSION TITLE
Proxy support for Request-FalconToken seems to be (silently) failing

### DISCUSSION THREAD

**Discussion by ypae** [CrowdStrike/psfalcon#111] [Q&A] [answered]
With PSFalcon 1.4.x, we have been successfully requesting $Falcon token with -Proxy parameter:

`Get-CsToken -ID <ID> -Secret <Secret> -Proxy <ProxyURL> -Debug
`

With PSFalcon 2.0.0, Request-FalconToken does not take -Proxy parameter and it silently fails to create $Falcon token:

`Request-FalconToken -ID <ID> -Secret <Secret> -Proxy <ProxyURL> -Debug
`
`Request-FalconToken -ID <ID> -Secret <Secret> -Debug
`

Any way to make PSFalcon 2.0 work under Proxy environment, please?

Also, when you are putting -Proxy option back in, please consider adding -ProxyCred option to explicitly provide proxy credential as part of the command line. 

Thanks,

Young-

**Comment by bk-cs** [comment, marked as answer]
Hi ypae,

Thanks for the issue! I think we spoke via email, as well.

There is no proxy handling in PSFalcon v2.0.0, because I never had any users using it in v1.4.2 so I didn't have any real-world testing. 

I recommend adding the proxy settings to your user profile and having PowerShell handle the proxy settings directly. Once you do that, then PSFalcon should work: [Using PowerShell behind a Proxy](http://woshub.com/using-powershell-behind-a-proxy/)

If that doesn't solve your problem, please let me know and I can review re-adding proxy support.
